azkaban-web-server/src/test/java/azkaban/flowtrigger/TriggerInstanceProcessorTest.java

```diff
@@ -82,11 +106,22 @@ public void testProcessSucceed() throws ExecutorManagerException {
   }
 
   @Test
-  public void testProcessTermination() throws ExecutorManagerException, InterruptedException {
+  public void testProcessTermination()
+      throws ExecutorManagerException, InterruptedException, IOException {
     final TriggerInstance triggerInstance = createTriggerInstance();
     processor.processTermination(triggerInstance);
     sleep(1000);
-    verify(emailer).sendEmail(any(), any(), any());
+    InputStream is = null;
+    try {
+      is = TriggerInstanceProcessorTest.class.getClassLoader()
+          .getResourceAsStream("emailTemplate/flowtriggerfailureemail.html");
+      final String emailHtml = IOUtils.toString(is, Charsets.UTF_8).trim();
+      assertThat(emailHtml).isEqualToIgnoringWhitespace(message.getBody());
```

Comment:
There'a already this in DefaultMailCreatorTest:

```java
  public static String read(final String file) throws Exception {
    final InputStream is = DefaultMailCreatorTest.class.getResourceAsStream(file);
    return IOUtils.toString(is, Charsets.UTF_8).trim();
  }
```

Hmm, `IOUtils.toString` doesn't close the stream? Then it should be changed to use try-with-resources to take care of closing the InputStream:

```java
  public static String read(final String file) throws Exception {
    try (final InputStream is = TriggerInstanceProcessorTest.class.getResourceAsStream(file)) {
      return IOUtils.toString(is, Charsets.UTF_8).trim();
    }
  }
```

Why not add a similar method in TriggerInstanceProcessorTest? Or better yet, combine them into some generic utility method that takes the parent class as an argument?

Any way, then the 10 lines of assertion code here would reduce to 2:

```java
    assertThat(TriggerInstanceProcessorTest.read("emailTemplate/flowtriggerfailureemail.html"))
        .isEqualToIgnoringWhitespace(this.message.getBody());
```

Reply:
updated, thanks